FullStorySegmentMiddleware/Classes/FullStoryMiddleware.m

```diff
@@ -146,16 +145,15 @@ - (NSDictionary *) getSuffixedProps: (NSDictionary *)properties {
         if ([@"@" isEqualToString:@(typeCode)]) {
             NSObject *obj = properties[key];
             
-            // if it's a NSDictionary: recurrsively get props, and we can not take underscore in key for nested objects
+            // if it's a NSDictionary: recurrsively get props, and we can not take underscore in key for nested objects, so we simply remove underscores
             // more info: https://help.fullstory.com/hc/en-us/articles/360020623234-FS-Recording-Client-API-Requirements
             if ([obj isKindOfClass:[NSDictionary class]]) {
                 [props setValue:[self getSuffixedProps:properties[key]]
                          forKey:[key stringByReplacingOccurrencesOfString:@"_" withString:@""]];
             } else if ([obj isKindOfClass:[NSArray class]]) {
-                // array of dicts? array of strings/ints/numbers? yuck!
                 NSArray *arr = (NSArray *) obj;
                 NSDictionary *dict = [self getDictionaryFromArrayObject:arr withKey:key];
-                [props addEntriesFromDictionary:dict];
+                [self appendToDictionary:props withKey:@"" andValue:dict];
```

Comment:
It would be worth adding a comment that the keys dictionary will be used (flattening). On a quick perusal of this code it looks like multiple arrays will overwrite each other.